tests/misc/test_access_links.py

```diff
@@ -0,0 +1,43 @@
+import pytest
+import subprocess
+import hashlib
+
+# This test is designed to verify the accessibility of the XOA deployment script
+#
+# Requirements:
+# - an XCP-ng host with xcp-ng-release >= 8.3.0.29
+
+@pytest.mark.parametrize("command", ["curl -fsSL", "wget -qO-"])
+@pytest.mark.parametrize("url", [
+    "https://xoa.io/deploy",
+    "https://updates.xcp-ng.org/trace",
+    "https://repo.vates.tech/README.txt"
+])
+def test_xoa_deploy(host, command, url):
+    """
+    Verifies that the specified URL responds correctly via the specified command
+    and compares the checksum of the downloaded content between local and remote.
+    """
+    COMMAND = f"{command} '{url}'"
+
+    # Download
+    local_result = subprocess.run(
+        COMMAND,
+        shell=True,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        text=True,
+    )
+    remote_result = host.ssh(COMMAND)
+
+    assert local_result.returncode == 0, (
+        f"Failed to fetch URL locally: {local_result.stderr}"
+    )
+
+    # Extract checksums
+    local_checksum = hashlib.sha256(local_result.stdout.split()[0].encode('utf-8')).hexdigest()
+    remote_checksum = hashlib.sha256(remote_result.split()[0].encode('utf-8')).hexdigest()
+
+    assert local_checksum == remote_checksum, (
```

Comment:
I was thinking about a more conventional "get checksums file from server" - but well here we don't have any so I guess that's about as much as we can test.

I'd still recommend to:
- first do the download that's the subject of the test
- then group the check, of which the local download is part, and label the whole part with an explanation of what we're doing